### DISCUSSION TITLE
How to set background color for Axes3D in Matplotlib?

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#753] [Q&A] [answered]
Hi Colour Team,

Sorry because this is not an issue, but I don't know where to ask.
In you guys's Chromaticity Diagram in 3D, I notice that you guys can set the background color of the 3D figure to be blank.
I tried many ways but cannot do the same thing, it still has some gray color like below
![image](https://user-images.githubusercontent.com/16201681/97337422-a4702880-1856-11eb-8e0d-158122211282.png)
How to make it all white, so I can only see three axis not the gray box anymore?

Thanks

**Comment by thomasmansencal** [comment]
Hi @hminle,

Did you try `axes.set_axis_off()` I haven't double-checked but I think that this is what does it!

https://github.com/colour-science/colour/blob/e65b38d1410c9834f124db56ae9e4eaa2467ad14/colour/plotting/common.py#L548

Cheers,

Thomas

**Comment by hminle** [comment]
Hi @KelSolaar , thank for your reply.
I want the axis on, but clear the background like the figure below:
![image](https://user-images.githubusercontent.com/16201681/97439067-164e7d80-18fc-11eb-8a1b-7d44497d328b.png)

It has no gray background box. I am not sure how.
I tried `axes.set_facecolor` but still not work

**Comment by thomasmansencal** [comment]
I take it you found out? :)

**Comment by hminle** [comment]
> I take it you found out? :)

No, I just gave up :)). Anyway, thank you a lot.

**Comment by thomasmansencal** [comment]
I will try to figure out over the weekend!

**Comment by hminle** [comment, marked as answer]
> I will try to figure out over the weekend!

Hi @KelSolaar , oh I've just found it.
It's here: https://stackoverflow.com/questions/29041326/3d-plot-with-matplotlib-hide-axes-but-keep-axis-labels
```
# Get rid of the panes
ax.w_xaxis.set_pane_color((1.0, 1.0, 1.0, 0.0))
ax.w_yaxis.set_pane_color((1.0, 1.0, 1.0, 0.0))
ax.w_zaxis.set_pane_color((1.0, 1.0, 1.0, 0.0))
```
Wow, just when I've been prepared to give up, I found it. lol

**Reply by yiping-git** [reply]
Update for later readers: `w_xaxis` attribute was **deprecated in Matplotlib 3.1** and will be removed in 3.8. Use `xaxis` instead:

```python
ax.xaxis.set_pane_color((1.0, 1.0, 1.0, 0.0))
ax.yaxis.set_pane_color((1.0, 1.0, 1.0, 0.0))
ax.zaxis.set_pane_color((1.0, 1.0, 1.0, 0.0))
```

**Comment by thomasmansencal** [comment]
Nice!
